### PR TITLE
test: use T.TempDir to create temporary test directory

### PR DESCRIPTION
**Pull request by Juneezee**
### Checklist

- [x] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md)
- [ ] My change requires a documentation update and I have done it
- [ ] I have added tests to cover my changes.

### Description
We can write less code by using the `T.TempDir` function from the `testing` package to create temporary directory. The directory created by `T.TempDir` is automatically removed when the test and all its subtests complete.

Reference: https://pkg.go.dev/testing#T.TempDir

### Screenshots (if appropriate):

---
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/2790)

### CODE REVIEW

**Review by notanatol — APPROVED** [review]

**Comment by Juneezee** [comment]
```
Error: error running commitlint
ensure-case: Unknown target case "sentence-case,start-case,pascal-case,upper-case"
TypeError: ensure-case: Unknown target case "sentence-case,start-case,pascal-case,upper-case"
    at toCase (/node_modules/@commitlint/rules/node_modules/@commitlint/ensure/lib/case.js:47:19)
    at Object.ensureCase [as case] (/node_modules/@commitlint/rules/node_modules/@commitlint/ensure/lib/case.js:18:25)
    at exports.bodyCase (/node_modules/@commitlint/rules/lib/body-case.js:14:33)
    at /node_modules/@commitlint/lint/lib/lint.js:98:40
    at Array.map (<anonymous>)
    at lint (/node_modules/@commitlint/lint/lib/lint.js:90:10)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async /src/action.js:130:19
    at async Promise.all (index 0)
    at async showLintResults (/src/action.js:[12](https://github.com/ethersphere/bee/runs/5178006933?check_suite_focus=true#step:6:12)8:25)
```

Possible issue with the commitlint action v2? Maybe we should update to v4 https://github.com/wagoid/commitlint-github-action/tags

File: pkg/localstore/localstore.go [hunk 1]
```diff
@@ -276,6 +276,8 @@ func New(path string, baseKey []byte, ss storage.StateStorer, o *Options, logger
 		// execute possible migrations
 		err = db.migrate(schemaName)
 		if err != nil {
+			_ = db.shed.Close()
```
Comment (mrekucci) [inline pkg/localstore/localstore.go:279]: You can use multierror here to also cover the error returned by `Close`.

**Comment by mrekucci** [comment]
> ```
> Error: error running commitlint
> ensure-case: Unknown target case "sentence-case,start-case,pascal-case,upper-case"
> TypeError: ensure-case: Unknown target case "sentence-case,start-case,pascal-case,upper-case"
>     at toCase (/node_modules/@commitlint/rules/node_modules/@commitlint/ensure/lib/case.js:47:19)
>     at Object.ensureCase [as case] (/node_modules/@commitlint/rules/node_modules/@commitlint/ensure/lib/case.js:18:25)
>     at exports.bodyCase (/node_modules/@commitlint/rules/lib/body-case.js:14:33)
>     at /node_modules/@commitlint/lint/lib/lint.js:98:40
>     at Array.map (<anonymous>)
>     at lint (/node_modules/@commitlint/lint/lib/lint.js:90:10)
>     at processTicksAndRejections (internal/process/task_queues.js:97:5)
>     at async /src/action.js:130:19
>     at async Promise.all (index 0)
>     at async showLintResults (/src/action.js:[12](https://github.com/ethersphere/bee/runs/5178006933?check_suite_focus=true#step:6:12)8:25)
> ```
> 
> Possible issue with the commitlint action v2? Maybe we should update to v4 https://github.com/wagoid/commitlint-github-action/tags

The problem is in the current setup of our commitlint config. The description should be just only on the first line.

File: pkg/localstore/localstore.go (hunk 1 above)
Comment (Juneezee) [reply pkg/localstore/localstore.go]: Fixed

**Review by mrekucci — APPROVED** [review]

**Review by notanatol — APPROVED** [review]

**Review by acud — APPROVED** [review]

**Pull request merged**